api/v1/db/json/notes-dao.js

```diff
@@ -73,24 +74,80 @@ const getNotes = query => new Promise((resolve, reject) => {
 });
 
 /**
- * @summary Return a specific pet by unique ID
+ * @summary Return a specific note by noteID
  * @function
- * @param {string} id Unique pet ID
- * @returns {Promise} Promise object represents a specific pet
+ * @param {string} noteID id of the note in the form: '{studentID}-{number}'
+ * @returns {Promise} Promise object represents a specific note
  */
-// const getPetById = id => new Promise((resolve, reject) => {
-//   try {
-//     const rawPets = appRoot.require('/tests/unit/mock-data.json').pets;
-//     const rawPet = _.find(rawPets, { ID: id });
-//     if (!rawPet) {
-//       resolve(undefined);
-//     } else {
-//       const serializedPet = SerializedPet(rawPet);
-//       resolve(serializedPet);
-//     }
-//   } catch (err) {
-//     reject(err);
-//   }
-// });
-
-module.exports = { getNotes };
+const getNoteByID = noteID => new Promise((resolve, reject) => {
+  try {
+    const studentID = noteID.split('-')[0];
+    const studentDirPath = `${dbDirectoryPath}/${studentID}`;
+
+    let rawNote;
+    try {
+      rawNote = JSON.parse(fs.readFileSync(`${studentDirPath}/${noteID}.json`, 'utf8'));
+    } catch (err) {
+      resolve(null);
+    }
+    rawNote.source = localSourceName;
+
+    const serializedNote = serializeNote(rawNote);
+    resolve(serializedNote);
+  } catch (err) {
+    reject(err);
+  }
+});
+
+/**
+ * @summary Create a new note
+ * @function
+ * @param body
+ * @returns {Promise} Promise object representing the new note
+ */
+const postNotes = body => new Promise((resolve, reject) => {
+  try {
+    const {
+      note, studentID, creatorID,
+    } = body;
+
+    // express-openapi does not correctly handle this default value so it must be specified manually
+    const permissions = body.permissions || 'advisor';
+    // ignore additional fields in context
+    const context = body.context ? {
+      contextType: body.context.contextType, contextID: body.context.contextID,
+    } : null;
+
+    const counter = fs.readFileSync(`${dbDirectoryPath}/${dbCounterFileName}`)
+      .toString().replace('\n', '');
+    const noteID = `${studentID}-${counter}`;
+
+    const newNote = {
+      id: noteID,
+      note,
+      studentID,
+      creatorID,
+      permissions,
+      context,
+    };
+    newNote.dateCreated = new Date().toISOString();
```

Comment:
No need to change, but just wanna mention that since `moment` package has been used already, we can also use `moment().toISOString()` to replace calling the built-in `Date` object.